File: leetcode/Math/leetcode-171/171.go
package math

func titleToNumber(s string) int {
	if len(s) == 0 {
		return 0
	}

	res := 0
	mul := 1

	for i := len(s) - 1; i >= 0; i-- {
		res += mul * (int)(s[i]-'A'+1)
		mul *= 26
	}
	return res
}

File: leetcode/Math/leetcode-171/171.java
class Solution {
    public int titleToNumber(String s) {
        if (s == null) {
            return 0;
        }

        int res = 0;
        int mul = 1;
        for (int i = s.length() - 1; i >= 0; i--) {
            res += mul * (s.charAt(i) - 'A' + 1);
            mul *= 26;
        }
        return res;
    }
}